pkgs/development/python-modules/triton/default.nix

```diff
@@ -30,33 +29,20 @@ let
 in
 buildPythonPackage rec {
   pname = "triton";
-  version = "2.1.0";
+  version = "3.0.0";
   pyproject = true;
 
   src = fetchFromGitHub {
-    owner = "openai";
+    owner = "triton-lang";
     repo = pname;
-    rev = "v${version}";
-    hash = "sha256-8UTUwLH+SriiJnpejdrzz9qIquP2zBp1/uwLdHmv0XQ=";
+    # latest branch commit from https://github.com/triton-lang/triton/commits/release/3.0.x/
+    rev = "91f24d87e50cb748b121a6c24e65a01187699c22";
+    hash = "sha256-L5KqiR+TgSyKjEBlkE0yOU1pemMHFk2PhEmxLdbbxUU=";
   };
 
-  patches =
-    [
-      # fix overflow error
-      (fetchpatch {
-        url = "https://github.com/openai/triton/commit/52c146f66b79b6079bcd28c55312fc6ea1852519.patch";
-        hash = "sha256-098/TCQrzvrBAbQiaVGCMaF3o5Yc3yWDxzwSkzIuAtY=";
-      })
-
-      # Upstream startded pinning CUDA version and falling back to downloading from Conda
-      # in https://github.com/triton-lang/triton/pull/1574/files#diff-eb8b42d9346d0a5d371facf21a8bfa2d16fb49e213ae7c21f03863accebe0fcfR120-R123
-      ./0000-dont-download-ptxas.patch
-    ]
-    ++ lib.optionals (!cudaSupport) [
-      # openai-triton wants to get ptxas version even if ptxas is not
-      # used, resulting in ptxas not found error.
-      ./0001-ptxas-disable-version-key-for-non-cuda-targets.patch
-    ];
+  # openai-triton wants to get ptxas version even if ptxas is not
+  # used, resulting in ptxas not found error.
+  patches = lib.optional (!cudaSupport) ./0000-ptxas-disable-version-key-for-non-cuda-targets.patch;
```

Comment:
- [ ] https://github.com/NixOS/nixpkgs/pull/328247/files#r1685461168
- [ ] `optionals`

Reply:
why optionals?

Reply:
Not a hill I'd die on, but people make the mistake of `optional c [x]` often enough I think we should just default to `optionals c [ x ]` instead